modules/flowable-cmmn-engine/src/test/java/org/flowable/cmmn/test/event/TaskCreatedEventTest.java

```diff
@@ -0,0 +1,90 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.flowable.cmmn.test.event;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.flowable.cmmn.api.runtime.CaseInstance;
+import org.flowable.cmmn.engine.test.CmmnDeployment;
+import org.flowable.cmmn.engine.test.FlowableCmmnTestCase;
+import org.flowable.common.engine.api.delegate.event.AbstractFlowableEventListener;
+import org.flowable.common.engine.api.delegate.event.FlowableEngineEventType;
+import org.flowable.common.engine.api.delegate.event.FlowableEvent;
+import org.flowable.common.engine.impl.event.FlowableEntityEventImpl;
+import org.flowable.task.api.Task;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Micha Kiener
+ */
+public class TaskCreatedEventTest extends FlowableCmmnTestCase {
+    protected CustomEventListener taskListener;
+
+    @Before
+    public void setUp() {
+        taskListener = new CustomEventListener();
+        cmmnEngineConfiguration.getEventDispatcher().addEventListener(taskListener, FlowableEngineEventType.TASK_CREATED);
+    }
+
+    @After
+    public void tearDown() {
+        if (taskListener != null) {
+            cmmnEngineConfiguration.getEventDispatcher().removeEventListener(taskListener);
+        }
+    }
+
+    @Test
+    @CmmnDeployment(resources = "org/flowable/cmmn/test/one-human-task-model.cmmn")
+    public void testCaseInstanceEvents() {
+        // start the case which will also need to throw the task started event
+        CaseInstance caseInstance = cmmnRuntimeService.createCaseInstanceBuilder()
+                .caseDefinitionKey("oneTaskCase")
+                .businessKey("business key")
+                .name("name")
+                .start();
+
+        Task task = cmmnTaskService.createTaskQuery()
+                .caseInstanceId(caseInstance.getId())
+                .singleResult();
+
+        FlowableEntityEventImpl caughtEvent = (FlowableEntityEventImpl) taskListener.caughtEvent;
+        assertThat(caughtEvent.getType()).isEqualTo(FlowableEngineEventType.TASK_CREATED);
+        assertThat(caughtEvent.getScopeId()).isEqualTo(caseInstance.getId());
+        assertThat(caughtEvent.getScopeDefinitionId()).isEqualTo(caseInstance.getCaseDefinitionId());
+        assertThat(caughtEvent.getSubScopeId()).isEqualTo(task.getId());
+        assertThat(caughtEvent.getScopeType()).isEqualTo(task.getScopeType());
```

Comment:
In place of lines 63-67 this alternative format can be used (I tend to favor using `extracting` and `contains` as I find it easier to read):
<pre>
        assertThat(caughtEvent).extracting(
                        FlowableEntityEventImpl::getType,
                        FlowableEntityEventImpl::getScopeId,
                        FlowableEntityEventImpl::getScopeDefinitionId,
                        FlowableEntityEventImpl::getSubScopeId,
                        FlowableEntityEventImpl::getScopeType)
                .containsExactly(
                        FlowableEngineEventType.TASK_CREATED,
                        caseInstance.getId(),
                        caseInstance.getCaseDefinitionId(),
                        task.getId(),
                        task.getScopeType()
                );
</pre>